mediator/esmFlds.F90

```diff
@@ -9,18 +9,29 @@ module esmflds
   ! Set components
   !-----------------------------------------------
 
-  integer, public, parameter  :: ncomps  = 8
-  integer, public, parameter  :: compmed = 1
-  integer, public, parameter  :: compatm = 2
-  integer, public, parameter  :: complnd = 3
-  integer, public, parameter  :: compocn = 4
-  integer, public, parameter  :: compice = 5
-  integer, public, parameter  :: comprof = 6
-  integer, public, parameter  :: compwav = 7
-  integer, public, parameter  :: compglc = 8
+  integer, public, parameter  :: compmed  = 1
+  integer, public, parameter  :: compatm  = 2
+  integer, public, parameter  :: complnd  = 3
+  integer, public, parameter  :: compocn  = 4
+  integer, public, parameter  :: compice  = 5
+  integer, public, parameter  :: comprof  = 6
+  integer, public, parameter  :: compwav  = 7
+  integer, public, parameter  :: compglc1 = 8
+  integer, public, parameter  :: ncomps   = 8
 
   character(len=*), public, parameter :: compname(ncomps) = &
-       (/'med','atm','lnd','ocn','ice','rof','wav','glc'/)
+       (/'med ',&
+         'atm ',&
+         'lnd ',&
+         'ocn ',&
+         'ice ',&
+         'rof ',&
+         'wav ',&
+         'glc '/)
+
+  integer, public, parameter :: max_icesheets = 1
+  integer, public :: compglc(max_icesheets) = (/compglc1/)
+  integer, public :: num_icesheets = 1
 
```

Comment:
It looks like at this point adding ice sheets would require several changes in the code, is the ultimate plan to make this a runtime option?

Reply:
Yes - this is just the skeleton framework that enables this. I envision a new attribute that cism and the mediator would use to determine the number of active ice sheets at run time. However, this needs to be sorted out with the cism folks and for now there are much bigger changes that need to be implemented in the cism cap in order to have multiple ice sheets.